ivy_tests/test_ivy/helpers/function_testing.py

```diff
@@ -264,6 +264,22 @@ def target_fn(instance, *args, **kwargs):
                     "the array in out argument does not contain same value as the"
                     " returned"
                 )
+        if test_flags.with_copy:
+            array_fn = ivy_backend.is_array
+            if "copy" in list(inspect.signature(target_fn).parameters.keys()):
+                kwargs["copy"] = True
+            first_array = ivy_backend.func_wrapper._get_first_array(
+                *args, array_fn=array_fn, **kwargs
+            )
+            ret_, ret_np_flat_ = get_ret_and_flattened_np_array(
+                fw,
+                target_fn,
+                *args,
+                test_compile=test_flags.test_compile,
+                **kwargs,
+            )
+            assert not np.may_share_memory(first_array, ret_)
```

Comment:
I think we need to assert that `ret_` is not a `tuple` here as is done in the frontend case below.